ui/packages/catalog-ui-search/src/main/webapp/component/query-advanced/query-advanced.view.js

```diff
@@ -161,7 +164,7 @@ module.exports = Marionette.LayoutView.extend({
     this.model.set({
       cql:
         this.options.isFormBuilder !== true
-          ? this.queryAdvanced.currentView.transformToCql()
+          ? this.queryAdvanced.currentView.transformToCql().replace(' null', '')
```

Comment:
When the IS EMPTY query is constructed, it is built as `("location" IS NULL null)`, but that's an invalid query, you need to remove ` null` so that you get `("location" IS NULL)` which is the actual query you want to send

Reply:
Would it be possible to change what this method is given so that we don't have to parse the string and can look at a data structure instead?

Reply:
Is there a way for the advanced query view to not put in the extra ' null' in the first place?

Reply:
I am looking into this, because the way the cql is built comes from the filter object which contains
{
   type
   property
   value
}
and the value is null for the EMPTY operator, when the cql is generated, it will append 'null' to the cql since it comes from the value field. However, there is a way to avoid this potentially and I am looking into it atm